src/main/ts/Part2Ex3Optional.ts

```diff
@@ -57,10 +64,13 @@ TODO: use Optional.from to implement the following DOM function
  */
 
 export const getNextSibling = (e: Element): Optional<ChildNode> => {
-  throw new Error("TODO");
+  return Optional.from(e ? e.nextSibling : null);
 };
 
 // TODO: use Optional.from to implement a similar wrapper for Element.getAttributeNode(string)
+export const getElementAttribute = (e: Element, attributeName: string): Optional<Attr> => {
+  return Optional.from(e && attributeName ? e.getAttributeNode(attributeName) : null)
```

Comment:
Similar to my previous comment this could be this instead:
```suggestion
  return Optional.from(e.getAttributeNode(attributeName));
```